postfixadmin: init at 3.3.10

globin opened the pull request:
###### Motivation for this change
Did not exist

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

globin committed:
postfixadmin: init at 3.3.9

globin committed:
postfixadmin: review additions

Co-authored-by: Linus Heckemann <git@sphalerite.org>

globin committed:
postfixadmin: fix db owner

globin committed:
postfixadmin: 3.3.9 -> 3.3.10

globin committed:
release-notes: add postfixadmin module addition

globin committed:
postfixadmin: stdenv.lib -> lib

Ma27 commented on nixos/modules/services/mail/postfixadmin.nix:
```diff
@@ -0,0 +1,198 @@
+  cfg = config.services.postfixadmin;
+  fpm = config.services.phpfpm.pools.postfixadmin;
+  localDB = cfg.database.host == "localhost";
+  user = if localDB then cfg.database.username else "nginx";
```
I think that the naming `database.username` is slightly misleading here. Apart from that, what's wrong with using `DynamicUser` in conjunction with `StateDirectory` (it's getting late, so perhaps I'm missing something :))?

Ma27 replied:
Just realized that this wouldn't work since the DynamicUser only exists when the service itself is running, but it has to be started *after* the database initialization is finished which is done in a different service, so this is the correct solution here.

Going to resolve the merge conflict locally and merge then.

Ma27 committed:
nixos/postfixadmin: fix eval & pin to PHP 7.4

Even though some PHP8 compat fixes seem to be in the release that's
packaged here, it seems as if there are still some minor issues[1].

[1] https://github.com/postfixadmin/postfixadmin/issues/395

siraben commented:
This PR caused NixOS Hydra evaluation to break (#138918) due to the introduction of `stdenv.lib`.
